### PR TITLE
chore: remove lambdas from config

### LINKED ISSUE #1684: Remove lambas from config
See discussion on https://github.com/instructlab/instructlab/pull/1682#pullrequestreview-2174717528.

```
src/instructlab/config/init.py:    default=lambda: DEFAULTS.DEFAULT_GGUF_MODEL,
src/instructlab/config/init.py:    default=lambda: DEFAULTS.TAXONOMY_DIR,
src/instructlab/data/generate.py:    default=lambda: DEFAULTS.DEFAULT_MODEL,
src/instructlab/data/generate.py:    default=lambda: DEFAULTS.TAXONOMY_DIR,
src/instructlab/data/generate.py:    default=lambda: DEFAULTS.DATASETS_DIR,
src/instructlab/lab.py:    default=lambda: cfg.DEFAULTS.CONFIG_FILE,
src/instructlab/model/download.py:    default=lambda: DEFAULTS.MODELS_DIR,
src/instructlab/model/test.py:    default=lambda: DEFAULTS.INTERNAL_DIR,
src/instructlab/model/test.py:    default=lambda: DEFAULTS.CHECKPOINTS_DIR,
src/instructlab/model/test.py:    default=lambda: DEFAULTS.DEFAULT_GGUF_MODEL,
src/instructlab/model/train.py:    default=lambda: DEFAULTS.DATASETS_DIR,
src/instructlab/model/train.py:    default=lambda: DEFAULTS.CHECKPOINTS_DIR,
src/instructlab/model/train.py:    default=lambda: DEFAULTS.INTERNAL_DIR,
src/instructlab/taxonomy/diff.py:    default=lambda: DEFAULTS.TAXONOMY_DIR,
```

### PR DESCRIPTION
**Pull request by nathan-weinberg**
**Issue resolved by this Pull Request:**
Resolves #1684 

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Comment by tiran** [comment]
Let's keep the lambdas in `configuration.py`. I have a pending change that will make `config.Config()` without additional settings work. It uses the lambdas.

It would be really helpful if you could remove all `default=` arguments from options that are provided by `config.yaml` and `ctx.default_map`. Subcommands should not need to access defaults from `instructlab.configuration`. Every default should come from the context.

**Comment by bjhargrave** [comment]
> It would be really helpful if you could remove all `default=` arguments

The default for `ilab taxonomy diff` breaks the command if you have specified a different folder during `ilab config init`.

```yaml
generate:
    taxonomy_path: /Users/hargrave/git/instructlab/taxonomy
```

```sh
➜  ilab taxonomy diff
Reading taxonomy failed with the following error: /Users/hargrave/Library/Application Support/instructlab/taxonomy
```

After removing default:

```sh
➜  ilab taxonomy diff
Taxonomy in /Users/hargrave/git/instructlab/taxonomy is valid :)
```

File: src/instructlab/configuration.py
Comment (RobotSail) [inline src/instructlab/configuration.py:263]: You can remove `Field` since this was only being used for the lambda functions.

```suggestion
    output_dir: StrictStr = DEFAULTS.DATASETS_DIR
```

File: src/instructlab/configuration.py
Comment (RobotSail) [inline src/instructlab/configuration.py:265]: Same comments here and and on `prompt_file` about the Field

**Review by RobotSail — APPROVED** [review]
LGTM once you can get this to pass tests. The reason the lambda existed in the first place was to have `ilab` avoid loading the defaults before running the tests. But now this behavior should no longer be the case since we're providing the `tmp_home_path` fixture and using `DEFAULTS._reset()`

**Review by leseb — COMMENTED** [review]
Blocked by https://github.com/instructlab/instructlab/pull/1697#issuecomment-2226365923? Should we close? Thanks!

**Comment by nathan-weinberg** [comment]
> Blocked by [#1697 (comment)](https://github.com/instructlab/instructlab/pull/1697#issuecomment-2226365923)? Should we close? Thanks!

Yup, let's close for now.

**Pull request closed without merging**
